Build axom@0.5.0 with spack@develop and xl@16.1.1.7 on lassen

I am making some progress, but I am now facing this error:

```
[ 42%] Building CXX object axom/sidre/CMakeFiles/sidre.dir/spio/interface/c_fortran/wrapIOManager.cpp.o
cd /usr/WS2/bernede1/gitlab-runner/builds/llnl-stack-e47f8392/spack/var/spack/stage/spack-stage-axom-0.5.0-6yc7op77pmemebj6rmwb53dvq42xxbim/spack-build-6yc7op7/axom/sidre && /usr/WS2/bernede1/gitlab-runner/builds/llnl-stack-e47f8392/spack/lib/spack/env/xl/xlc++ -x c++ -DAXOM_DEBUG -I/usr/WS2/bernede1/gitlab-runner/builds/llnl-stack-e47f8392/spack/var/spack/stage/spack-stage-axom-0.5.0-6yc7op77pmemebj6rmwb53dvq42xxbim/spack-build-6yc7op7/include -I/usr/WS2/bernede1/gitlab-runner/builds/llnl-stack-e47f8392/spack/var/spack/stage/spack-stage-axom-0.5.0-6yc7op77pmemebj6rmwb53dvq42xxbim/spack-src/src -I/usr/WS2/bernede1/gitlab-runner/builds/llnl-stack-e47f8392/spack/var/spack/stage/spack-stage-axom-0.5.0-6yc7op77pmemebj6rmwb53dvq42xxbim/spack-build-6yc7op7/lib/fortran -I/usr/workspace/radiuss/install/radiuss-stack/[padded-to-128-chars]/linux-rhel7-ppc64le/xl-16.1.1.7/camp-0.1.0-emm6dzjdgx5wlnsjarsi4grygz57dkua/include -I/usr/workspace/radiuss/install/radiuss-stack/[padded-to-128-chars]/linux-rhel7-ppc64le/xl-16.1.1.7/raja-0.13.0-ubrabga3n3p5uz3vyghmuoitrdxdrmbz/include -I/usr/WS1/radiuss/install/radiuss-stack/[padded-to-122-chars]/linux-rhel7-ppc64le/xl-16.1.1.7/conduit-0.7.2-kgmbmxwmwblt3gr2abnbw675x2r4cokb/include -I/usr/WS1/radiuss/install/radiuss-stack/[padded-to-122-chars]/linux-rhel7-ppc64le/xl-16.1.1.7/conduit-0.7.2-kgmbmxwmwblt3gr2abnbw675x2r4cokb/include/conduit -I/usr/WS2/bernede1/gitlab-runner/builds/llnl-stack-e47f8392/spack/var/spack/stage/spack-stage-axom-0.5.0-6yc7op77pmemebj6rmwb53dvq42xxbim/spack-src/src/thirdparty -I/usr/workspace/radiuss/install/radiuss-stack/[padded-to-128-chars]/linux-rhel7-ppc64le/xl-16.1.1.7/umpire-4.1.2-4icm6ls7rdid4wblvgy6jtjl2533dwal/include -I/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-2020.08.19/include -I/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-rolling-release/include -I/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-2020.08.19/lib/XL -I/usr/workspace/radiuss/install/radiuss-stack/[padded-to-128-chars]/linux-rhel7-ppc64le/xl-16.1.1.7/conduit-0.7.2-kgmbmxwmwblt3gr2abnbw675x2r4cokb/include -I/usr/workspace/radiuss/install/radiuss-stack/[padded-to-128-chars]/linux-rhel7-ppc64le/xl-16.1.1.7/conduit-0.7.2-kgmbmxwmwblt3gr2abnbw675x2r4cokb/include/conduit -I/usr/WS1/radiuss/install/radiuss-stack/[padded-to-122-chars]/linux-rhel7-ppc64le/xl-16.1.1.7/hdf5-1.8.22-hl7fyxcpeeor3hrre6gf5lqjuzmvnofb/include -std=c++1y      -g -fPIC -qsmp=omp -pthread -std=c++14 -o CMakeFiles/sidre.dir/spio/interface/c_fortran/wrapIOManager.cpp.o -c /usr/WS2/bernede1/gitlab-runner/builds/llnl-stack-e47f8392/spack/var/spack/stage/spack-stage-axom-0.5.0-6yc7op77pmemebj6rmwb53dvq42xxbim/spack-src/src/axom/sidre/spio/interface/c_fortran/wrapIOManager.cpp
1 warning generated.
In file included from /usr/WS2/bernede1/gitlab-runner/builds/llnl-stack-e47f8392/spack/var/spack/stage/spack-stage-axom-0.5.0-6yc7op77pmemebj6rmwb53dvq42xxbim/spack-src/src/axom/inlet/LuaReader.cpp:16:
In file included from /usr/WS2/bernede1/gitlab-runner/builds/llnl-stack-e47f8392/spack/var/spack/stage/spack-stage-axom-0.5.0-6yc7op77pmemebj6rmwb53dvq42xxbim/spack-src/src/axom/inlet/LuaReader.hpp:18:
/usr/WS2/bernede1/gitlab-runner/builds/llnl-stack-e47f8392/spack/var/spack/stage/spack-stage-axom-0.5.0-6yc7op77pmemebj6rmwb53dvq42xxbim/spack-src/src/thirdparty/sol/sol.hpp:492:16: error: no type named 'make_index_sequence' in namespace 'std'
                typedef std::make_index_sequence<sizeof...(Args)> indices;
                        ~~~~~^                     
[...]
```

Briefly looking at the code, I thought there could be an `#include <utility>` missing before line 492 (it is present later in the file). But the fact it didn't show up before makes me think I'm wrong.

[lassen-blueos_3_ppc64le_ib_p9-xl@16.1.1.7.cmake.txt](https://github.com/LLNL/axom/files/6870196/lassen-blueos_3_ppc64le_ib_p9-xl%4016.1.1.7.cmake.txt)

--- a/scripts/spack/packages/axom/package.py
+++ b/scripts/spack/packages/axom/package.py
@@ -3,11 +3,11 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
-from spack import *
-
 import os
 import socket
 from os.path import join as pjoin
+
+from spack import *
 
 
 def get_spec_path(spec, package_name, path_replacements={}, use_bin=False):
@@ -279,15 +279,20 @@ class Axom(CachedCMakePackage, CudaPackage):
                 # Fix for working around CMake adding implicit link directories
                 # returned by the BlueOS compilers to link executables with
                 # non-system default stdlib
-                _gcc_prefix = "/usr/tce/packages/gcc/gcc-4.9.3/lib64"
-                if os.path.exists(_gcc_prefix):
-                    _gcc_prefix2 = pjoin(
-                        _gcc_prefix,
-                        "gcc/powerpc64le-unknown-linux-gnu/4.9.3")
-                    _link_dirs = "{0};{1}".format(_gcc_prefix, _gcc_prefix2)
+                _roots = ["/usr/tce/packages/gcc/gcc-4.9.3",
+                          "/usr/tce/packages/gcc/gcc-4.9.3/gnu"]
+                _subdirs = ["lib64",
+                            "lib64/gcc/powerpc64le-unknown-linux-gnu/4.9.3"]
+                _existing_paths = []
+                for root in _roots:
+                    for subdir in _subdirs:
+                        _curr_path = pjoin(root, subdir)
+                        if os.path.exists(_curr_path):
+                            _existing_paths.append(_curr_path)
+                if _existing_paths:
                     entries.append(cmake_cache_string(
                         "BLT_CMAKE_IMPLICIT_LINK_DIRECTORIES_EXCLUDE",
-                        _link_dirs))
+                        ";".join(_existing_paths)))
 
         return entries
 

